app/src/shared/web3/web3Poller/web3Poller.js

```diff
@@ -0,0 +1,185 @@
+        this.emitter.on(event, handler)
+    }
+
+    unsubscribe(event: Event, handler: Handler) {
+        this.emitter.removeListener(event, handler)
+    }
+
+    startWeb3Poll = () => {
+        this.clearWeb3Poll()
+        this.web3PollTimeout = setTimeout(this.pollWeb3, WEB3_POLL_INTERVAL)
+    }
+
+    pollWeb3 = () => (
+        this.fetchWeb3Account()
+            .then(this.startWeb3Poll, this.startWeb3Poll)
+    )
+
+    startEthereumNetworkPoll = () => {
+        this.clearEthereumNetworkPoll()
+        this.ethereumNetworkPollTimeout = setTimeout(this.pollEthereumNetwork, NETWORK_POLL_INTERVAL)
+    }
+
+    pollEthereumNetwork = () => (
+        this.fetchChosenEthereumNetwork()
+            .then(this.startEthereumNetworkPoll, this.startEthereumNetworkPoll)
+    )
+
+    clearWeb3Poll = () => {
+        if (this.web3PollTimeout) {
+            clearTimeout(this.web3PollTimeout)
+            this.web3PollTimeout = null
+        }
+    }
+
+    clearEthereumNetworkPoll = () => {
+        if (this.ethereumNetworkPollTimeout) {
+            clearTimeout(this.ethereumNetworkPollTimeout)
+            this.ethereumNetworkPollTimeout = null
+        }
+    }
+
+    clearPendingTransactionsPoll = () => {
+        if (this.pendingTransactionsPollTimeout) {
+            clearTimeout(this.pendingTransactionsPollTimeout)
+            this.pendingTransactionsPollTimeout = null
+        }
+    }
+
+    fetchWeb3Account = () => (
+        this.web3.getDefaultAccount()
+            .then((account) => {
+                this.handleAccount(account)
+                // needed to avoid warnings about creating promise inside a handler
+                // if any other web3 actions are dispatched.
+                return Promise.resolve()
+            }, (err) => {
+                if (this.account) {
+                    this.account = null
+                    this.emitter.emit(events.ACCOUNT_ERROR, err)
+                }
+            })
+    )
+
+    handleAccount = (account: string) => {
+        const next = account
+
+        const didChange = !!(this.account && next && !areAddressesEqual(this.account, next))
+        const didDefine = !!(!this.account && next)
+
+        // Check current provider so that account event is not sent prematurely
+        // (ie. wait for user to approve access to Metamask)
+        if (this.web3.currentProvider !== null && (didDefine || didChange)) {
+            this.account = next
+            this.emitter.emit(events.ACCOUNT, next)
+        }
+    }
+
+    fetchChosenEthereumNetwork = () => (
+        this.web3.getEthereumNetwork()
+            .then((network) => {
+                this.handleNetwork(network.toString())
+            }, () => {
+                this.web3.getDefaultAccount()
+                    .catch((err) => {
+                        if (this.account !== null) {
+                            this.emitter.emit(events.NETWORK_ERROR, err)
+                        }
```

Comment:
The network polling is actually useless atm since changing the network will refresh the page. This might need refactoring anyway at some point, leaving it for now.